File: Javascript/258.add_digits [1].js
/**
 * @param {number} num
 * @return {number}
 */
var addDigits = function(num) {
    while(true){
        const digitSum = num
        .toString()
        .split('')
        .reduce((acc, dig) => acc + parseInt(dig), 0)

        if(digitSum < 10)
            return digitSum
        
        num = digitSum
    }

    return -1
};

File: CPP [0001-0500]/258.add-digits [1].cpp
class Solution {
private:
    int getDigitSum(int num){
        int sum = 0;
        while(num > 0)
        {
            sum += num % 10;
            num /= 10;
        }

        return sum;
    }

public:
    // T.C.=O(??), S.C.=O(1)
    int addDigits(int num) {
        while(true)
        {
            int digitSum = getDigitSum(num);

            if(digitSum < 10)
                return digitSum;
            
            num = digitSum;
        }

        return -1;
    }
};